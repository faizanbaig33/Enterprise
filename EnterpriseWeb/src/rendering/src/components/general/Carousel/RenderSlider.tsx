import { Field, Placeholder } from '@sitecore-jss/sitecore-jss-nextjs';

import React, { ReactElement } from 'react';
import { SliderWrapper } from 'src/helpers/SliderWrapper';
import { Caption } from 'src/helpers/Caption';
import { PlaceholderData } from '@sitecore-jss/sitecore-jss/layout';
import { useTheme } from 'lib/context/ThemeContext';
import { CarouselTheme } from './Carousel.theme';

type childItem = {
  fields: {
    caption: Field<string>;
  };
  placeholders: PlaceholderData;
};

type RenderingProps = {
  slidesData: childItem[];
};
type RenderSliderProps = {
  sliderSettings: { [key: string]: any };
} & RenderingProps;

export const RenderSlider = ({ slidesData, sliderSettings }: RenderSliderProps): ReactElement => {
  const { themeData } = useTheme(CarouselTheme);
  return (
    <>
      <SliderWrapper sliderSettings={sliderSettings}>
        {slidesData.map((childItem: any, index: number) => {
          return (
            <Placeholder
              key={index}
              name="components"
              rendering={childItem}
              render={(childComponents) => {
                return childComponents.map((component, index) => (
                  <div key={index}>
                    <div className="[&_.section-grid]:px-0">{component}</div>
                    <Caption {...childItem.fields} classes={themeData.classes.slideCaption} />
                  </div>
                ));
              }}
            />
          );
        })}
      </SliderWrapper>
    </>
  );
};
