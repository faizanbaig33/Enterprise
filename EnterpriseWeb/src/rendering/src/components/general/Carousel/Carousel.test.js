// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
import 'lib/jest/match-media-mock';
// Local
import Carousel from './Carousel';
import defaultData from './Carousel.mock-data';

it('renders correctly', () => {
  const component = snapshot(Carousel, { componentProps: defaultData });
  hasDataComponent(component, 'general/carousel');
});
