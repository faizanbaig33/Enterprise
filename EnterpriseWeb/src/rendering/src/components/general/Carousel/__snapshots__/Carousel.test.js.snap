// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/carousel"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12"
      >
        <div
          class="items-top font-sans   text-theme-text text-sm-m md:text-m font-heavy mb-xxs "
        >
          <h2>
            Optional heading
          </h2>
          
        </div>
        <div
          class="font-sans text-theme-text text-sm-s md:text-s font-medium mb-s "
        >
          <h3>
            Optional description of carousel. Here’s what’s going on below.
          </h3>
        </div>
        <div
          class="slick-slider numbered-pagination slick-initialized"
          dir="ltr"
        >
          <span
            class="slick-arrow slick-prev text-theme-text hover:text-theme-text"
            style="display: block;"
          >
            <span />
          </span>
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="true"
                class="slick-slide slick-cloned"
                data-index="-1"
                style="width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="sc-jss-placeholder-error"
                  >
                    A rendering error occurred: 
                    Cannot read properties of null (reading 'prefetch')
                    .
                  </div>
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="sc-jss-placeholder-error"
                  >
                    A rendering error occurred: 
                    Cannot read properties of null (reading 'prefetch')
                    .
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="1"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="sc-jss-placeholder-error"
                  >
                    A rendering error occurred: 
                    Cannot read properties of null (reading 'prefetch')
                    .
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="2"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="sc-jss-placeholder-error"
                  >
                    A rendering error occurred: 
                    Cannot read properties of null (reading 'prefetch')
                    .
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="3"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="sc-jss-placeholder-error"
                  >
                    A rendering error occurred: 
                    Cannot read properties of null (reading 'prefetch')
                    .
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide slick-cloned"
                data-index="4"
                style="width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="sc-jss-placeholder-error"
                  >
                    A rendering error occurred: 
                    Cannot read properties of null (reading 'prefetch')
                    .
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide slick-cloned"
                data-index="5"
                style="width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="sc-jss-placeholder-error"
                  >
                    A rendering error occurred: 
                    Cannot read properties of null (reading 'prefetch')
                    .
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide slick-cloned"
                data-index="6"
                style="width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="sc-jss-placeholder-error"
                  >
                    A rendering error occurred: 
                    Cannot read properties of null (reading 'prefetch')
                    .
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide slick-cloned"
                data-index="7"
                style="width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="sc-jss-placeholder-error"
                  >
                    A rendering error occurred: 
                    Cannot read properties of null (reading 'prefetch')
                    .
                  </div>
                </div>
              </div>
            </div>
          </div>
          <span
            class="slick-arrow slick-next text-theme-text hover:text-theme-text"
            style="display: block;"
          >
            <span />
          </span>
          <ul
            class="mt-xs text-center"
            style="display: block;"
          >
            <li
              class="slick-active"
            >
              <span
                class=""
              >
                1
                 / 
                4
              </span>
            </li>
            <li
              class=""
            >
              <span
                class="hidden"
              >
                2
                 / 
                4
              </span>
            </li>
            <li
              class=""
            >
              <span
                class="hidden"
              >
                3
                 / 
                4
              </span>
            </li>
            <li
              class=""
            >
              <span
                class="hidden"
              >
                4
                 / 
                4
              </span>
            </li>
          </ul>
        </div>
        <div
          class="justify-center mt-m mb-s flex items-start md:flex-row md:items-center"
        >
          <a
            class="flex w-fit items-center whitespace-nowrap rounded-lg border-4 border-white bg-white px-m py-[9px] font-sans text-button font-heavy text-black hover:border-black hover:bg-black hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray font-heavy text-secondary"
            href="/"
            role="link"
            tabindex="0"
            target="_blank"
            title=""
          >
            Learn More
            <span
              class="ml-xxs"
            />
            <span
              class="sr-only"
            >
               (Opens in a new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
